src/Data/PQueue/Prio/Internals.hs

```diff
@@ -650,6 +662,17 @@ foldrWithKeyF_ f fCh ts0 z0 = case ts0 of
     fCh' (Succ (BinomTree k a ts) tss) z =
       f k a (fCh ts (fCh tss z))
 
+-- | Unordered monoidal fold on a binomial forest.
+foldMapWithKeyF_ :: Monoid m => (k -> a -> m) -> (rk k a -> m) -> BinomForest rk k a -> m
```

Comment:
Why are you defining standalone functions here instead of also uisng a typeclass?

Reply:
Just going by the pattern the module was using before. I don't know which way (if either) performs better. We can evaluate that ... once we have benchmarks. Do you know of any benchmarks out there?

Reply:
I don't unfortunately, I've only done a few very basic benchmarks (`insert` and `deleteMin`) a while ago.

I think we should at least be consistent between `Data.PQueue` and `Data.PQueue.Prio`.

Reply:
Fine. I'll go with the class way for now and we can revisit later. I want to pack in a few more features (including deciding how to expose internals) and get a release out.